### PR TITLE
Network stats getter test

### PR DESCRIPTION
**Pull request by alvarocabanas**
Refactor network_stats_getter and system_cpu_reader to have similar structure with open and parse and add tests.

### CODE REVIEW

File: test/integration/metrics_test.go [hunk 1]
```diff
@@ -246,7 +246,7 @@ func TestExitedContainersWithoutTTL(t *testing.T) {
 		"/",
 		raw.NewCgroupV1PathParser(),
 		raw.NewPosixSystemCPUReader(),
-		raw.NetDevNetworkStatsGetter{},
+		raw.NewNetDevNetworkStatsGetter(),
```
Comment (sigilioso) [inline test/integration/metrics_test.go:249]: This is out of the scope of this PR but I think we could use some kind of helper to build the testing fetcher (we are repeating the `raw.NewCgroupsV1Fetcher(...` call a lot.

**Review by sigilioso — APPROVED** [review]
LGTM 👍

File: test/integration/metrics_test.go (hunk 1 above)
Comment (alvarocabanas) [reply test/integration/metrics_test.go:249]: Yes it's true

**Review by marcsanmi — APPROVED** [review]
LGTM!

**Pull request merged**